prowler

```diff
@@ -370,13 +407,21 @@ if __name__ == "__main__":
             args.organizations_role,
         )
     elif provider == "azure":
-        audit_info = azure_provider_set_session(subscriptions)
+        audit_info = azure_provider_set_session(
+            subscriptions, az_cli_auth, sp_env_auth, browser_auth, managed_entity_auth
+        )
 
     # Check if custom output filename was input, if not, set the default
     if not output_filename:
-        output_filename = (
-            f"prowler-output-{audit_info.audited_account}-{output_file_timestamp}"
-        )
+        if provider == "aws":
+            output_filename = (
+                f"prowler-output-{audit_info.audited_account}-{output_file_timestamp}"
+            )
+        elif provider == "azure":
+            if audit_info.identity.domain:
+                output_filename = f"prowler-output-{audit_info.identity.domain}-{output_file_timestamp}"
+            else:
+                output_filename = f"prowler-output-{'-'.join(audit_info.identity.tenant_ids)}-{output_file_timestamp}"
```

Comment:
Can we move this logic to outputs?

Reply:
Right now we need to keep it here since we are parsing here input options